lms/djangoapps/discussion/rest_api/tests/test_api_v2.py

```diff
@@ -561,6 +572,140 @@ def test_thread_un_abuse_flag_for_moderator_role(
+        self.register_thread_votes_response("test_thread")
+        self.register_thread()
+        data = {"voted": new_vote_status}
+        result = update_thread(request1, "test_thread", data)
+        assert result["voted"] == new_vote_status
+
+        vote_update_func_calls = self.get_mock_func_calls("update_thread_votes")
+        last_function_args = (
+            vote_update_func_calls[-1] if vote_update_func_calls else None
+        )
+
+        if current_vote_status == new_vote_status:
+            assert last_function_args is None
+        else:
+            if vote_update_func_calls:
+                assert last_function_args[1]["value"] == (
+                    "up" if new_vote_status else "down"
+                )
+                params = {
+                    "thread_id": "test_thread",
+                    "value": "up" if new_vote_status else "down",
+                    "user_id": str(user1.id),
+                    "course_id": str(self.course.id),
+                }
+                self.check_mock_called_with("update_thread_votes", -1, **params)
+            else:
+                params = {
+                    "thread_id": "test_thread",
+                    "user_id": str(user1.id),
+                    "course_id": str(self.course.id),
+                }
+                self.check_mock_called_with("delete_thread_vote", -1, **params)
+            event_name, event_data = mock_emit.call_args[0]
+            assert event_name == "edx.forum.thread.voted"
+            assert event_data == {
+                "undo_vote": (not new_vote_status),
+                "url": "",
+                "target_username": self.user.username,
+                "vote_value": "up",
+                "user_forums_roles": [FORUM_ROLE_STUDENT],
+                "user_course_roles": [],
+                "commentable_id": "original_topic",
+                "id": "test_thread",
+            }
+
+    @ddt.data(*itertools.product([True, False], [True, False], [True, False]))
+    @ddt.unpack
+    def test_vote_count(self, current_vote_status, first_vote, second_vote):
+        """
+        Tests vote_count increases and decreases correctly from the same user
+        """
+        # setup
+        starting_vote_count = 0
+        user, request = self.create_user_with_request()
+        if current_vote_status:
+            self.register_get_user_response(user, upvoted_ids=["test_thread"])
+            starting_vote_count = 1
+        self.register_thread_votes_response("test_thread")
+        self.register_thread(overrides={"votes": {"up_count": starting_vote_count}})
+
+        # first vote
+        data = {"voted": first_vote}
+        result = update_thread(request, "test_thread", data)
+        self.register_thread(overrides={"voted": first_vote})
+        assert result["vote_count"] == (1 if first_vote else 0)
+
+        # second vote
+        # TODO: Add inline comment
```

Comment:
we can update the todo msg to something meaningful

Reply:
I forgot to add the message. I'll update it.